types/wordpress__components/visually-hidden/index.d.ts

```diff
@@ -2,7 +2,7 @@ import { ComponentType, ReactNode } from 'react';
 import { WordPressComponentProps } from '../ui/context/wordpress-component';
 
 declare namespace VisuallyHidden {
-    type Props = WordPressComponentProps<ReactNode, 'div'>;
+    type Props = WordPressComponentProps<{}, 'div'> & { children?: ReactNode };
```

Comment:
```suggestion
    type Props = WordPressComponentProps<{ children?: ReactNode }, 'div'>;
```

Reply:
That would not allow children since **all** props from the first type parameter are excluded. Did you try this suggestion locally with https://github.com/DefinitelyTyped/DefinitelyTyped/pull/56210?

Reply:
`WordPressComponentProps` absolutely does an intersection between `P`, the omitted section you noted (but I think you're misreading how it works, as it omits any props from P that would be present on the implicit props derived from T, it doesn't omit anything from P) and then merges it with the `as` prop.

Here is a link to the `wordpress-component.d.ts` file and line on your branch where this is happening, copied here for posterity:

https://github.com/eps1lon/DefinitelyTyped/blob/fix/implicit-children/wordpress__components/types/wordpress__components/ui/context/wordpress-component.d.ts#L14

```ts
export type WordPressComponentProps<P, T extends ElementType, IsPolymorphic extends boolean = true> = P &
    Omit<ComponentPropsWithRef<T>, 'as' | keyof P | 'children'> &
    (IsPolymorphic extends true
        ? {
              as?: T | keyof JSX.IntrinsicElements;
          }
        : { as?: never });
```

Note the `P &` at the very start of the type expression on the right hand side of the assignment.

I've taken a look at the React 18 PR you shared and can't see anything that would invalidate this intersection. The only thing I can see that might need to change for React 18 in this package (aside from this fix, which is valid) is that maybe `children` no longer needs to be omitted from `ComponentPropsWithRef` (potentially, I haven't looked into the React 18 types specifically, I'm just deducing that from the other changes being made in that PR).

Reply:
Ah it's not formatted using prettier. Will check that out.